rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -543,6 +546,177 @@ void addSignatureMultipleInputsPartiallyValid() throws Exception {
+        BridgeEventLogger eventLogger = new BridgeEventLoggerImpl(bridgeMainNetConstants, arrowheadActivations, eventLogs, signatureCache);
+
+        // Build prev btc tx
+        BtcTransaction prevTx = new BtcTransaction(btcParams);
+        TransactionOutput prevOut = new TransactionOutput(btcParams, prevTx, Coin.FIFTY_COINS, federation.getAddress());
+        prevTx.addOutput(prevOut);
+
+        // Build btc tx to be signed
+        BtcTransaction btcTx = new BtcTransaction(btcParams);
+        btcTx.addInput(prevOut).setScriptSig(createBaseInputScriptThatSpendsFromTheFederation(federation));
+        TransactionOutput output = new TransactionOutput(btcParams, btcTx, Coin.COIN, new BtcECKey().toAddress(btcParams));
+        btcTx.addOutput(output);
+
+        // Save btc tx to be signed
+        SortedMap<Keccak256, BtcTransaction> pegoutWaitingForSignatures = new TreeMap<>();
+
+        final Keccak256 rskTxHash = createHash3(1);
+        pegoutWaitingForSignatures.put(rskTxHash, btcTx);
+        when(provider.getPegoutsWaitingForSignatures())
+            .thenReturn(pegoutWaitingForSignatures);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withBridgeConstants(bridgeMainNetConstants)
+            .withProvider(provider)
+            .withEventLogger(eventLogger)
+            .withBtcLockSenderProvider(new BtcLockSenderProvider())
+            .withActivations(arrowheadActivations)
+            .build();
+
+        BtcECKey privateKeyToSignWith = BtcECKey.fromPrivate(BigInteger.valueOf(150));
+        Optional<FederationMember> federationMember = federation.getMemberByBtcPublicKey(privateKeyToSignWith);
+        assertTrue(federationMember.isPresent());
+        FederationMember federationMemberToSignWith = federationMember.get();
+        BtcECKey federatorBtcPubKey = federationMemberToSignWith.getBtcPublicKey();
+
+        Script inputScript = btcTx.getInputs().get(0).getScriptSig();
+        List<ScriptChunk> chunks = inputScript.getChunks();
+        byte[] program = chunks.get(chunks.size() - 1).data;
+        Script redeemScript = new Script(program);
+        Sha256Hash sigHash = btcTx.hashForSignature(0, redeemScript, BtcTransaction.SigHash.ALL, false);
+        BtcECKey.ECDSASignature sig = privateKeyToSignWith.sign(sigHash);
+        List derEncodedSigs = Collections.singletonList(sig.encodeToDER());
+
+        // Act
+        bridgeSupport.addSignature(federationMemberToSignWith.getBtcPublicKey(), derEncodedSigs, rskTxHash.getBytes());
+
+        // Assert
+        commonAssertLogs(eventLogs);
+        assertTopics(3, eventLogs);
+
+        String derivedRskAddress = ByteUtil.toHexString(federationMemberToSignWith.getRskPublicKey().getAddress());
+        assertEvent(eventLogs, 0, BridgeEvents.ADD_SIGNATURE.getEvent(), new Object[]{rskTxHash.getBytes(), derivedRskAddress}, new Object[]{federatorBtcPubKey.getPubKey()});
+    }
```

Comment:
these two tests should be parameterized. We can pass the `derivedRskAddress` as a hardcoded value since we are getting the pub key from a hardcoded priv key.

Reply:
Agree

Reply:
Nice. Good one.